pkg/volume/hostpath/host_path.go

```diff
@@ -151,6 +151,15 @@ func (plugin *hostPathPlugin) Recycle(pvName string, spec *volume.Spec, eventRec
 			Path: spec.PersistentVolume.Spec.HostPath.Path,
 		},
 	}
+	if spec.PersistentVolume.Spec.NodeAffinity != nil {
```

Comment:
add unit tests for these?

Reply:
This is just for prevent nil panic, hostPath localpv must have NodeAffinity, will not be nil.